site/src/components/Popover/Popover.tsx

```diff
@@ -41,32 +38,40 @@ const PopoverContext = createContext<PopoverContextValue | undefined>(
   undefined,
 );
 
-export interface PopoverProps {
-  children: ReactNode | ((popover: PopoverContextValue) => ReactNode); // Allows inline usage
+type BasePopoverProps = {
+  children: ReactNode;
   mode?: TriggerMode;
-  isDefaultOpen?: boolean;
-}
+};
 
-export const Popover: FC<PopoverProps> = ({
-  children,
-  mode,
-  isDefaultOpen,
-}) => {
+type UncontrolledPopoverProps = BasePopoverProps & {
+  open?: undefined;
+  onOpenChange?: undefined;
+};
+
+type ControlledPopoverProps = BasePopoverProps & {
+  open: boolean;
+  onOpenChange: (open: boolean) => void;
+};
+
+export type PopoverProps = UncontrolledPopoverProps | ControlledPopoverProps;
+
+export const Popover: FC<PopoverProps> = (props) => {
   const hookId = useId();
-  const [isOpen, setIsOpen] = useState(isDefaultOpen ?? false);
-  const triggerRef = useRef<HTMLElement>(null);
+  const [open, setOpen] = useState(false);
+  const triggerRef: TriggerRef = useRef(null);
+  const isControlled = props.open !== undefined;
 
   const value: PopoverContextValue = {
-    isOpen,
-    setIsOpen,
     triggerRef,
     id: `${hookId}-popover`,
-    mode: mode ?? "click",
+    mode: props.mode ?? "click",
+    open: isControlled ? props.open : open,
+    setOpen: isControlled ? props.onOpenChange : setOpen,
```

Comment:
Could these be simplified via nullish coalescing? Should also remove the need for the `isControlled` variable

```tsx
open: props.open ?? open,
setOpen: props.onOpenChange ?? setOpen
```